```
error[E0599]: no method named `raw_read` found for type `servo_arc::Arc<lock_api::rwlock::RwLock<parking_lot::RawRwLock, ()>>` in the current scope
  --> components/style/shared_lock.rs:77:18
   |
77 |         self.arc.raw_read();
   |                  ^^^^^^^^
error[E0599]: no method named `raw_write` found for type `servo_arc::Arc<lock_api::rwlock::RwLock<parking_lot::RawRwLock, ()>>` in the current scope
  --> components/style/shared_lock.rs:90:18
   |
90 |         self.arc.raw_write();
   |                  ^^^^^^^^^
   |
   = help: did you mean `try_write`?
error[E0599]: no method named `raw_unlock_read` found for type `servo_arc::Arc<lock_api::rwlock::RwLock<parking_lot::RawRwLock, ()>>` in the current scope
   --> components/style/shared_lock.rs:112:29
    |
112 |         unsafe { self.0.arc.raw_unlock_read() }
    |                             ^^^^^^^^^^^^^^^
    |
    = help: did you mean `force_unlock_read`?
error[E0599]: no method named `raw_unlock_write` found for type `servo_arc::Arc<lock_api::rwlock::RwLock<parking_lot::RawRwLock, ()>>` in the current scope
   --> components/style/shared_lock.rs:127:29
    |
127 |         unsafe { self.0.arc.raw_unlock_write() }
    |                             ^^^^^^^^^^^^^^^^
    |
    = help: did you mean `force_unlock_write`?
   Compiling script_traits v0.0.1 (file:///home/travis/build/servo/servo/components/script_traits)
error: aborting due to 4 previous errors
For more information about this error, try `rustc --explain E0599`.
error: Could not compile `style`.
warning: build failed, waiting for other jobs to finish...
error: build failed
The command "./mach test-unit" exited with 101.
```

--- a/components/layout/Cargo.toml
+++ b/components/layout/Cargo.toml
@@ -28,7 +28,7 @@ malloc_size_of = { path = "../malloc_size_of" }
 msg = {path = "../msg"}
 net_traits = {path = "../net_traits"}
 new-ordered-float = "1.0"
-parking_lot = "0.5"
+parking_lot = "0.6"
 profile_traits = {path = "../profile_traits"}
 range = {path = "../range"}
 rayon = "1"

--- a/components/layout_thread/Cargo.toml
+++ b/components/layout_thread/Cargo.toml
@@ -33,7 +33,7 @@ malloc_size_of = { path = "../malloc_size_of" }
 metrics = {path = "../metrics"}
 msg = {path = "../msg"}
 net_traits = {path = "../net_traits"}
-parking_lot = "0.5"
+parking_lot = "0.6"
 profile_traits = {path = "../profile_traits"}
 range = {path = "../range"}
 rayon = "1"

--- a/components/script/Cargo.toml
+++ b/components/script/Cargo.toml
@@ -70,7 +70,7 @@ msg = {path = "../msg"}
 net_traits = {path = "../net_traits"}
 num-traits = "0.1.32"
 offscreen_gl_context = {version = "0.20", features = ["serde"]}
-parking_lot = "0.5"
+parking_lot = "0.6"
 phf = "0.7.18"
 profile_traits = {path = "../profile_traits"}
 ref_filter_map = "1.0.1"

--- a/components/style/Cargo.toml
+++ b/components/style/Cargo.toml
@@ -51,7 +51,7 @@ num-integer = "0.1.32"
 num-traits = "0.1.32"
 new-ordered-float = "1.0"
 owning_ref = "0.3.3"
-parking_lot = "0.5"
+parking_lot = "0.6"
 precomputed-hash = "0.1.1"
 rayon = "1"
 selectors = { path = "../selectors" }

--- a/components/style/shared_lock.rs
+++ b/components/style/shared_lock.rs
@@ -11,6 +11,8 @@ use parking_lot::RwLock;
 use servo_arc::Arc;
 use std::cell::UnsafeCell;
 use std::fmt;
+#[cfg(feature = "servo")]
+use std::mem;
 #[cfg(feature = "gecko")]
 use std::ptr;
 use str::{CssString, CssStringWriter};
@@ -74,7 +76,7 @@ impl SharedRwLock {
     /// Obtain the lock for reading (servo).
     #[cfg(feature = "servo")]
     pub fn read(&self) -> SharedRwLockReadGuard {
-        self.arc.raw_read();
+        mem::forget(self.arc.read());
         SharedRwLockReadGuard(self)
     }
 
@@ -87,7 +89,7 @@ impl SharedRwLock {
     /// Obtain the lock for writing (servo).
     #[cfg(feature = "servo")]
     pub fn write(&self) -> SharedRwLockWriteGuard {
-        self.arc.raw_write();
+        mem::forget(self.arc.write());
         SharedRwLockWriteGuard(self)
     }
 
@@ -107,9 +109,9 @@ pub struct SharedRwLockReadGuard<'a>(AtomicRef<'a, SomethingZeroSizedButTyped>);
 #[cfg(feature = "servo")]
 impl<'a> Drop for SharedRwLockReadGuard<'a> {
     fn drop(&mut self) {
-        // Unsafe: self.lock is private to this module, only ever set after `raw_read()`,
+        // Unsafe: self.lock is private to this module, only ever set after `read()`,
         // and never copied or cloned (see `compile_time_assert` below).
-        unsafe { self.0.arc.raw_unlock_read() }
+        unsafe { self.0.arc.force_unlock_read() }
     }
 }
 
@@ -122,9 +124,9 @@ pub struct SharedRwLockWriteGuard<'a>(AtomicRefMut<'a, SomethingZeroSizedButType
 #[cfg(feature = "servo")]
 impl<'a> Drop for SharedRwLockWriteGuard<'a> {
     fn drop(&mut self) {
-        // Unsafe: self.lock is private to this module, only ever set after `raw_write()`,
+        // Unsafe: self.lock is private to this module, only ever set after `write()`,
         // and never copied or cloned (see `compile_time_assert` below).
-        unsafe { self.0.arc.raw_unlock_write() }
+        unsafe { self.0.arc.force_unlock_write() }
     }
 }
 

--- a/tests/unit/style/Cargo.toml
+++ b/tests/unit/style/Cargo.toml
@@ -15,7 +15,7 @@ app_units = "0.6"
 cssparser = "0.24.0"
 euclid = "0.18"
 html5ever = "0.22"
-parking_lot = "0.5"
+parking_lot = "0.6"
 rayon = "1"
 serde_json = "1.0"
 selectors = {path = "../../../components/selectors"}

